### PR TITLE
template: modify AKS from VM to VMSS node pool

### PR DESCRIPTION
**Pull request by 3quanfeng**
This PR modifies the node pool type in the deployment template.

### CODE REVIEW

File: deploy/azuredeploy.json [hunk 1]
```diff
@@ -603,7 +603,10 @@
                         "osType": "Linux",
                         "storageProfile": "ManagedDisks",
                         "vnetSubnetID": "[variables('kubernetesSubnetId')]",
-                        "dnsPrefix": "agents"
+                        "dnsPrefix": "agents",
+                        "maxPods": 30,
+                        "type": "VirtualMachineScaleSets",
```
Comment (akshaysngupta) [inline deploy/azuredeploy.json]: Are the remaining params: `dnsPrefix/maxPods/enableAutoScaling` manadatory ?

Comment (3quanfeng) [reply deploy/azuredeploy.json]: maxPods should be required when "enableAutoScaling" is true

Comment (akshaysngupta) [reply deploy/azuredeploy.json]: if not, lets get rid of them ?

Comment (3quanfeng) [reply deploy/azuredeploy.json]: done

**Comment by SychevIgor** [comment]
any plans to merge this PR?

**Pull request closed without merging**